### ISSUE TITLE
Error: Expected type "text/html" but got "image/jpeg"

### ISSUE THREAD

**Issue by actuallymentor** [stevenvachon/broken-link-checker#65]
Hello!

I'm loving this package, but seem to be running into this error a lot. Specifically when checking png and jpg clickable links.

Is this a known issue?

**Comment by stevenvachon** [comment]
Yes, it's caused by a navigable link to an image. The checker expects it to be an html document.

**Comment by actuallymentor** [comment]
Is there a way to ignore the type? I'm happy with a status 200 regardless of mime match...

**Comment by stevenvachon** [comment]
The mime match is necessary for not parsing a resource as html. There's currently no nice way of avoiding the error, though. You can create a custom logger with the programmatic API; using a file extension check before logging an error in the `page` event callback.

**Comment by actuallymentor** [comment]
I worked around it by setting:
```js 
excludedKeywords: [ '.png', '.jpg' ]
```

Which is reasonably elegant.

It's a shame that testing image links will not work, but as far as things go it's a minor thing.

**Comment by stevenvachon** [comment]
Oh yeah, forgot about that option. Those keywords, however, will block domains, usernames, passwords and directories that contain such.

I don't know if this will be fixed in 0.8.0 or not, but it will eventually happen.

**Comment by actuallymentor** [comment]
Does the option take Regex? Making one for a url ending in .whatever would be trivial.

**Comment by stevenvachon** [comment]
Currently, no. PR welcome, though.

**Comment by stevenvachon** [comment]
Leaving this open until it's correctly fixed.

**Comment by actuallymentor** [comment]
I decided to make a (an npm module)[https://www.npmjs.com/package/check-broken-links] to check for links in my tests.

While it is super versatile, this package ended up needing to much hacking for my particular needs. Might still come back and do a PR since your work taught me a lot by reading it :)

**Comment by stevenvachon** [comment]
Sure, whatever works. This project will have a lot of major changes in 8.x and more after. It's a deep rabbit hole.
